lib/editing/editor.dart

```diff
@@ -60,6 +60,13 @@ abstract class Editor {
 
   bool get hasFocus;
 
+  /**
+   * Fired when a mouse is clicked. You can preventDefault the event to signal
+   * that the editor should do no further handling.  Only implemented for
+   * codemirror, returns `null` for ace editor and comid.
+   */
+  Stream get onMouseDown;
```

Comment:
Can you parameterize this type? `Stream<MouseEvent>`, here, and in the subclasses.